vllm/model_executor/layers/quantization/modelopt.py

```diff
@@ -1580,6 +1580,20 @@ def get_fused_moe_quant_config(
         ):
             return None
 
+        if (
+            self.flashinfer_moe_backend == FlashinferMoeBackend.CUTLASS
+            and not layer.moe_config.dp_size > 1
+        ):
+            return FusedMoEQuantConfig.make(
+                quant_dtype="nvfp4_skip_quantization",
```

Comment:
Why not just leave quant_dtype as `None`?

Reply:
> here

It seems like it would hit that `else` even if the quant_dtype is "nvfp4_skip_quantization"?  Could you change it to check for non-empty scales instead of just falling thru to the bf16 case?

I think `quant_dtype` should be limited to represent actual types and not types + some flag.  Whenever torch finally supports fp4 types it would be good to remove the `str` option from `quant_dtype` entirely.  Or switch `quant_dtype` to use `vllm.scalar_type | None`.  (Obviously not for this PR).